src/Visualization/Utility/SelectionPolygon.h

```diff
@@ -71,6 +74,10 @@ class SelectionPolygon : public Geometry2D
             const PointCloud &input, const ViewControl &view);
     std::shared_ptr<PointCloud> CropPointCloudInPolygon(
             const PointCloud &input, const ViewControl &view);
+    std::shared_ptr<TriangleMesh> CropTriangleMeshInRectangle(
+            const TriangleMesh &input, const ViewControl &view);
+    std::shared_ptr<TriangleMesh> CropTriangleMeshInPolygon(
+            const TriangleMesh &input, const ViewControl &view);
```

Comment:
Same as previous comment, maybe template functions.

Reply:
I think it is fine. Let's leave it for now.
Plus, I can't think of any other use cases. These two are all that we need.